python/lsst/ts/observatory/control/base_tcs.py

```diff
@@ -952,6 +955,20 @@ async def _offset(self, offset_cmd):
         await asyncio.sleep(self.tel_settle_time)
         self.log.debug("Done")
 
+    def ready_to_take_data(self):
+        """Return a future object that will be done when the ATCS is ready to
+        take data or raise an exception if something wrong happens while trying
+        to determine the condition of the system.
+        """
+        if self._ready_to_take_data_future.done():
+            self._ready_to_take_data_future = asyncio.Future()
+            self._ready_to_take_data_task = asyncio.create_task(
+                self._ready_to_take_data()
+            )
+            return self._ready_to_take_data_future
+        else:
+            return self._ready_to_take_data_future
```

Comment:
You can get rid of the `else` block and simply return `self._ready_to_take_data_future` in a line at the same indentation as the `if` statement.

Reply:
good catch